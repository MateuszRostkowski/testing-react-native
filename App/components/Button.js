import React from 'react'
import { TouchableOpacity, Text, StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  button: {
    backgroundColor: '#D22322',
    paddingVertical: 10,
    paddingHorizontal: 45,
    alignItems: 'center',
    marginHorizontal: 20,
  },
  buttonLoading: {
    backgroundColor: '#E58E8D',
  },
  text: {
    fontWeight: '500',
    fontSize: 18,
    color: '#fff',
  },
})

export const Button = ({ text, onPress, onLongPress, disabled = false, style = {} }) => (
  <TouchableOpacity
    onPress={onPress}
    onLongPress={onLongPress}
    style={[styles.button, disabled && styles.buttonLoading, style]}
    disabled={disabled}>
    <Text style={styles.text} testID="buttonText">
      {text}
    </Text>
  </TouchableOpacity>
)
