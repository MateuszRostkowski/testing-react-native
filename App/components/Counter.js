import React, { useState } from 'react'
import { Button } from './Button'

export const Counter = () => {
  const [count, setCount] = useState(0)

  const increment = () => {
    setCount(state => state + 1)
  }

  const decrement = () => {
    setCount(state => state - 1)
  }

  return <Button text={count} onPress={increment} onLongPress={decrement} />
}
