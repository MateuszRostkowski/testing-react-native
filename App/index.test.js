import React from 'react'
import { render, fireEvent } from '@testing-library/react-native'
import { Counter } from './components/Counter'

test('Render corectly', () => {
  const { toJSON } = render(<Counter />)

  expect(toJSON()).toMatchSnapshot()
})

test('Test counter', async () => {
  const { getByTestId } = render(<Counter />)
  const buttonText = getByTestId('buttonText')

  fireEvent.press(buttonText)
  expect(buttonText.props.children).toBe(1)
  fireEvent.press(buttonText)
  expect(buttonText.props.children).toBe(2)
})
