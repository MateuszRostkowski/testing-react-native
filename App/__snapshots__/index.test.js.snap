// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render corectly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#D22322",
      "marginHorizontal": 20,
      "opacity": 1,
      "paddingHorizontal": 45,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 18,
        "fontWeight": "500",
      }
    }
    testID="buttonText"
  >
    0
  </Text>
</View>
`;
